persistence/src/main/scala/com/arangodb/async/internal/ArangoExecutorAsyncDestructor.scala

```diff
@@ -19,7 +19,7 @@ package com.arangodb.async.internal
 import com.arangodb.internal.velocystream.VstCommunication
 import za.co.absa.commons.reflect.ReflectionUtils
 
-object ArangoExecutorAsyncExtractor {
+object ArangoExecutorAsyncDestructor {
```

Comment:
What is the reason for this rename? Extractor seems ok, Destructor seems a bit confusing. To destruct usually means to destroy.

Reply:
To me `FooExtractor` reads as it extracts Foo from something. While in fact in this case it actually extracts something from Foo. And that confused me when I looked at the code after a few months of writing it.

Reply:
I was thinking of _destructurer_, _decomposer_... Unfortunately there is no naming convention or at least any recommendation how to name this thing in Scala world. In most code that I've seen it doesn't have its own as it's usually just an _unapply_ method in a companion object. Since I couldn't reuse the same naming approach for a library class and had do pick another name, I ended up with the word _destructor_ using vocabulary from this Kotlin related article - https://opencredo.com/blogs/the-destructor-pattern/